test/jdk/java/awt/TrayIcon/ActionCommand/ActionCommand.java

```diff
@@ -44,15 +58,19 @@ public class ActionCommand {
     TrayIcon icon;
     ExtendedRobot robot;
 
-    boolean actionPerformed = false;
-    Object actionLock = new Object();
-    String actionCommand = null;
+    volatile boolean actionPerformed = false;
+    volatile String actionCommand = null;
+    final Object actionLock = new Object();
+
     static boolean isMacOS = false;
 
     public static void main(String[] args) throws Exception {
-        if (! SystemTray.isSupported()) {
-            System.out.println("SystemTray not supported on the platform under test. " +
-                    "Marking the test passed");
+        if (Platform.isOnWayland()) {
```

Comment:
Perhaps, each `if` statement could be standalone… without the `else`. For both conditions `Platform.isOnWayland()` and `!SystemTray.isSupported()`, an exception is thrown, which means the test effectively finished. The following statements would be executed if and only if, no exception was thrown, therefore the indentation of the code could be reduced and the code would also look cleaner.

Reply:
Sure, this is better, updated.